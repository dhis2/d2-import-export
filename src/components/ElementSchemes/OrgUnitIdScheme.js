import React, { useContext } from 'react';
import i18n from '@dhis2/d2-i18n';

import { SchemeContext } from '../../contexts/';
import { Select } from '../Select';

const OrgUnitIdScheme = ({ selected, setSelected }) => {
    const { OrgUnitId } = useContext(SchemeContext);
    return (
        <Select
            name="OrgUnitIdScheme"
            label={i18n.t('Organisation unit ID scheme')}
            options={OrgUnitId}
            selected={selected}
            setValue={setSelected}
            dense
        />
    );
};

export { OrgUnitIdScheme };
