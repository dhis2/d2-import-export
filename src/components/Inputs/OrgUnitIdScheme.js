import i18n from '@dhis2/d2-i18n'
import React from 'react'
import {
    RadioGroup,
    RadioGroupContainer,
    RadioGroupLabel,
} from '../FinalFormComponents/RadioGroup'

export const OPTION_UID = { value: 'UID', label: i18n.t('Uid') }
export const OPTION_CODE = { value: 'CODE', label: i18n.t('Code') }
export const OPTION_NAME = { value: 'NAME', label: i18n.t('Name') }

export const ORG_UNIT_ID_SCHEME_DEFAULT_OPTIONS = [
    OPTION_UID,
    OPTION_CODE,
    OPTION_NAME,
]
export const ORG_UNIT_ID_SCHEME_KEY = 'orgUnitIdScheme'
export const ORG_UNIT_ID_SCHEME_DEFAULT_VALUE = OPTION_UID.value

const orgUnitIdSchemeLabel = i18n.t('Org unit id scheme')
export const OrgUnitIdScheme = ({ options }) => (
    <RadioGroup
        name={ORG_UNIT_ID_SCHEME_KEY}
        label={orgUnitIdSchemeLabel}
        options={options}
    />
)

OrgUnitIdScheme.propTypes = {
    options: RadioGroup.propTypes.options,
}

OrgUnitIdScheme.defaultProps = {
    options: ORG_UNIT_ID_SCHEME_DEFAULT_OPTIONS,
}

export const OrgUnitIdSchemeLoading = () => (
    <RadioGroupContainer>
        <RadioGroupLabel>{orgUnitIdSchemeLabel}</RadioGroupLabel>
        {i18n.t('Loading organisation unit id scheme options...')}
    </RadioGroupContainer>
)
