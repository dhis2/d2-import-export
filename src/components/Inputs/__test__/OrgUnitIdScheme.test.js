import React from 'react'
import renderer from 'react-test-renderer'
import { OrgUnitIdScheme } from '../OrgUnitIdScheme'

jest.mock('react-final-form', () => ({
    useField: jest.fn(() => ({
        input: {
            name: 'Name',
            value: '',
            onChange: jest.fn(),
        },
        meta: {
            touched: false,
            error: '',
        },
    })),
}))

describe('Input component - OrgUnitIdScheme', () => {
    it('should render correctly', () => {
        const file = renderer.create(<OrgUnitIdScheme />).toJSON()

        expect(file).toMatchSnapshot()
    })
})
