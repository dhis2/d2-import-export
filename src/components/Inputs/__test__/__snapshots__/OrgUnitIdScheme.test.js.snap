// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - OrgUnitIdScheme should render correctly 1`] = `
<Select
  data-test-id="input-org-unit-id-scheme"
  label="Org unit id scheme"
  name="orgUnitIdScheme"
  options={
    Array [
      Object {
        "label": "Uid",
        "value": "UID",
      },
      Object {
        "label": "Code",
        "value": "CODE",
      },
      Object {
        "label": "Name",
        "value": "NAME",
      },
    ]
  }
/>
`;
